.github/workflows/scan-from-dockerhub-report-issue.yaml

```diff
@@ -0,0 +1,38 @@
+name: Scan from published image and report vulnerabilities
+
+on:
+  workflow_call:
+    inputs:
+      image-name:
+        description: "The published image name to be scanned."
+        required: true
+        type: string
+      report-vulnerabilities:
+        description: "Whether to report security vulnerabilities through Github issues."
+        required: false
+        default: false
+        type: boolean
+      severity:
+        description: "Comma separated list of severities of vulnerabilities to scanned for and displayed"
+        required: false
+        type: string
+        default: "UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL"
+jobs:
+  scan:
+    # uses: canonical/charmed-kubeflow-workflows/.github/workflows/scan-from-published-image.yaml@KF-6331-add-scan-reusable-workflow
+    uses: ./.github/workflows/scan-from-published-image.yaml
+    secrets: inherit
+    with:
+      image-name: ${{ inputs.image-name }}
+      report-vulnerabilities: ${{ inputs.report-vulnerabilities }}
+      severity: ${{ inputs.severity }}
+
+  # Leaving it here in case we ever want to enable scan and reports on_merge
+  report-vulnerability:
+    needs: scan
+    uses: ./.github/workflows/report-vulnerability-in-gh.yaml
+    secrets: inherit
+    if: ${{ always() && (needs.scan.result == 'failure') }}
```

Comment:
I don't get when we want this to run. Shouldn't this depend on `inputs.report-vulnerabilities`?

Reply:
When a trivy scan fails using the action, the job itself will fail with exit code 1.

We want to report issues whenever that happens, that's why this says if the `scan` job fails, run this. You have a point on if we want to keep `report-vulnerabilities`, and I think we can remove it entirely. The reason being, in this workflow we always assume this input is going to be set to `true`, the reason why I left it here was because at the beginning I thought we could re-use more code than we actually do.

If you agree, I will remove the `report-vulnerabilities` and all related logic.

Reply:
Removed, please let me know what you think. Also check the latest [run](https://github.com/DnPlas/kubeflow-rocks/actions/runs/11278881988/job/31368399088) to confirm this is working as expected.

Reply:
I think we don't the `if` anymore too. Other than that, I agree with what you described and final solution looks good.

Reply:
We still need it, otherwise the next job is skipped because the previous one failed.